### PR TITLE
Code cleanup and initial cut of docs

### PR DESCRIPTION
**Pull request by myitcv**
* Stop using dot imports of `github.com/myitcv/gopherjs/react` in examples
* Initial cut of tidy up of core `github.com/myitcv/gopherjs/react` package files
* Initial cut of docs for core React elements

### CODE REVIEW

File: react/react.go [hunk 1]
```diff
@@ -9,15 +9,30 @@ import (
...
+	reactCompShouldComponentUpdate     = "shouldComponentUpdate"
+	reactCompComponentDidMount         = "componentDidMount"
+	reactCompComponentWillReceiveProps = "componentWillReceiveProps"
+	reactCompComponentWillMount        = "componentWillMount"
+	reactCompComponentWillUnmount      = "componentWillUnmount"
+	reactCompRender                    = "render"
+
+	reactCreateElement = "createElement"
+	reactCreateClass   = "createClass"
+	reactDomRender     = "render"
```
Comment (shurcooL) [inline react/react.go]: DOM is an acronym for Document Object Model. According to Go style, they should have a consistent case, so this should be:

```
reactDOMRender
```

Source: https://github.com/golang/go/wiki/CodeReviewComments#initialisms.

File: react/examples/hellomessage/hello_message.go [hunk 2]
```diff
@@ -1,11 +1,11 @@
 package hellomessage
 
 import (
-	. "github.com/myitcv/gopherjs/react"
+	r "github.com/myitcv/gopherjs/react"
```
Comment (shurcooL) [inline react/examples/hellomessage/hello_message.go]: 👍

File: react/react.go [hunk 3]
```diff
@@ -9,15 +9,30 @@ import (
...
+
+	nestedProps            = "_props"
+	nestedState            = "_state"
+	nestedComponentWrapper = "__ComponentWrapper"
 )
 
 var react = js.Global.Get("React")
 var reactDom = js.Global.Get("ReactDOM")
```
Comment (shurcooL) [inline react/react.go]: This should've been named `reactDOM` too, for the same reason.

**Comment by myitcv** [comment]
@shurcooL appreciate you taking a look; this is still totally work in progress (I try to indicate that via the `WIP` in the subject) so I'll give you a shout when this is ready. Thanks again

**Comment by myitcv** [comment]
@shurcooL I merged the update to vendored code in a separate PR (on the basis that is largely mechanical) and I rebased this PR so it only contains the tidy up and initial cut of docs. Any feedback appreciated.

File: react/text_area.go [hunk 4]
```diff
@@ -2,18 +2,22 @@ package react
...
+// TextAreaPropsDef defines the properties for the <textarea> element
 type TextAreaPropsDef struct {
 	*BasicHTMLElement
 
-	Placeholder  string `js:"placeholder"`
-	Value        string `js:"value"`
-	DefaultValue string `js:"defaultValue"`
+	Placeholder        string `js:"placeholder"`
+	Value              string `js:"value"`
+	DefatextareatValue string `js:"defatextareatValue"`
```
Comment (shurcooL) [inline react/text_area.go]: Is this correct?

File: react/a.go [hunk 5]
```diff
@@ -2,17 +2,21 @@ package react
...
 	underlying *js.Object
 }
 
+// APropsDef defines the properties for the <a> element
 type APropsDef struct {
 	*BasicHTMLElement
 
 	Target string `js:"target"`
-	Href   string `js:"href"`
+	Href   string `js:"aef"`
```
Comment (shurcooL) [inline react/a.go]: Is this change intentional?

File: react/text_area.go (hunk 4 above)
Comment (myitcv) [inline react/text_area.go]: Thanks - remnant of a rather over-ambitious `sed`

File: react/a.go (hunk 5 above)
Comment (myitcv) [inline react/a.go]: Thanks - remnant of a rather over-ambitious `sed`
